src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java

```diff
@@ -89,13 +98,19 @@ public void onActivate(ActivateEvent event, EntityRef entity) {
 
             // Check to see if player has other team's flag
             if (baseTeamComponent.team.equals(LASUtils.RED_TEAM)
+                    && heldItem.hasComponent(BlockItemComponent.class)
                     && heldItem.getComponent(BlockItemComponent.class).blockFamily.getURI().toString().equals(LASUtils.BLACK_FLAG_URI)) {
                 redScore++;
+                inventoryManager.removeItem(player, player, heldItem, true);
+                worldProvider.setBlock(new Vector3i(LASUtils.CENTER_BLACK_BASE_POSITION.x, LASUtils.CENTER_BLACK_BASE_POSITION.y + 1, LASUtils.CENTER_BLACK_BASE_POSITION.z), blockManager.getBlock(LASUtils.BLACK_FLAG_URI));
             }
 
             if (baseTeamComponent.team.equals(LASUtils.BLACK_TEAM)
+                    && heldItem.hasComponent(BlockItemComponent.class)
                     && heldItem.getComponent(BlockItemComponent.class).blockFamily.getURI().toString().equals(LASUtils.RED_FLAG_URI)) {
                 blackScore++;
+                inventoryManager.removeItem(player, player, heldItem, true);
+                worldProvider.setBlock(new Vector3i(LASUtils.CENTER_RED_BASE_POSITION.x, LASUtils.CENTER_RED_BASE_POSITION.y + 1, LASUtils.CENTER_RED_BASE_POSITION.z), blockManager.getBlock(LASUtils.RED_FLAG_URI));
             }
```

Comment:
I don’t see where the flag is removed from the losing player’s inventory and reset to it’s base (the losing player might also be halfway through running with the other player’s flag)

Have you not implemented this or am I missing something?

Reply:
Correct, this hasn't been implemented yet